Userland/Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -271,15 +271,15 @@ static inline i32* get_component(Macroblock& block, unsigned component)
  * macroblocks that share the chrominance data. Next two iterations (assuming that
  * we are dealing with three components) will fill up the blocks with chroma data.
  */
-static bool build_macroblocks(JPGLoadingContext& context, Vector<Macroblock>& macroblocks, u32 hcursor, u32 vcursor)
+static ErrorOr<void> build_macroblocks(JPGLoadingContext& context, Vector<Macroblock>& macroblocks, u32 hcursor, u32 vcursor)
 {
     for (unsigned component_i = 0; component_i < context.component_count; component_i++) {
         auto& component = context.components[component_i];
 
         if (component.dc_destination_id >= context.dc_tables.size())
-            return false;
+            return Error::from_string_literal("component.dc_destination_id >= context.dc_tables.size()");
         if (component.ac_destination_id >= context.ac_tables.size())
-            return false;
+            return Error::from_string_literal("component.ac_destination_id >= context.ac_tables.size()");
```

Comment:
Same note here about using an actual error message and not just copying the conditional